platform/qdcontainer/container.go

```diff
@@ -115,6 +118,28 @@ func CheckContainerEngineMemory() {
 
 // GetContainerClient getContainerClient returns a docker client.
 func GetContainerClient() *client.Client {
+	cli, err := command.NewDockerCli()
+	if err != nil {
+		log.Fatal("couldn't create Docker CLI: ", err)
+	}
+	opts := flags.NewClientOptions()
+	if err := cli.Initialize(opts); err != nil {
+		log.Fatal("couldn't initialize Docker CLI: ", err)
+	}
+	apiClient := cli.Client()
+	// apiClient always implements the Client interface
+	// client/interface_stable.go
+	// var _ APIClient = &Client{}
+	c, ok := apiClient.(*client.Client)
+	if !ok {
+		log.Warn("Initialising light version of docker client, some docker configuration could be missing")
+		return GetLightContainerClient()
+	}
```

Comment:
I think that api client not inheriting from client is a logic error and should asserted instead of providing a workaround. I suggest returning `apiClient.(*client.Client)` directly and cutting out `GetLightContainerClient` entirely. The type assertion will not change under us as long as we don't release untested builds.